### PR TITLE
[Term Entry] C++ vector.erase() #4987

### LINKED ISSUE #4987: [Term Entry] C++ vector .erase()
### Reason/inspiration (optional)

"We would like a new entry on the `.erase()` term entry in Cpp under `Vector` concept. The entry should go in a new file under `content/cpp/concepts/vectors/terms/erase/erase.md.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept.
- An `Example` section that provides an example demonstrating the concept in use.

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Son0-lory**
### Description

Added a new term entry for the *.erase()* method on vectors in C++ language.

### Issue Solved

Close #4987

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    for (unsigned i=0; i<room.size(); ++i)
+        std::cout << ' ' << room[i];
+    std::cout << '\n';
+
+    return 0;
+}
+```
+The output of the above code is :
+```shell
+Rooms available: 1 2 4 5 6 7
```
Comment (Son0-lory) [inline content/cpp/concepts/vectors/terms/erase/erase.md:94]: Rooms available: 1 2 3 4 5 6 7

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Vectors'
+  - 'Arrays'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:14]:
```suggestion
---
Title: '.erase()'
Description: 'Removes a single element or a range of elements from a vector.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Arrays'
  - 'Data Structures'
  - 'Vectors'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:14]: Arranged the tags in an alphabetical manner

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+The **.erase()** function removes a single element or a range of elements from a specific position of the `vector`.
+When an element is remove with *.erase()* method, the size of the `vector` decrease. Therefore the elements after the ones deleted are moved to a new position.
+The position of the elements to remove are identified with `.begin()` and/or `.end()` functions.
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:18]:
```suggestion
The **`.erase()`** function removes a single element or a range of elements from a specific position in the `vector`. When an element is removed with the `.erase()` function, the size of the `vector` decreases, and the elements after the deleted ones are shifted to fill the gap. The positions of the elements to remove are identified with `.begin()` and/or `.end()` functions.
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:18]: Modifications:
1. the first instance of the term should be in **bold** and `backticks`
2. Reframed the sentence for better clarity, it had some spelling mistakes
3. Every time we use a term, always put it in backticks: `erase()`, `.end()`, etc
4. We can either call it as a function or a method, so I changed the instance of method to function in this sentence.

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+## Syntax
+
+```  
+vector.erase(position);  
+vector.erase(starting_position, ending_position);   
+```
+The firsr syntax removes a single element.
+The second syntax removes a range of elements.
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:27]:
```suggestion
```  
To remove a single element from the vector:

```
vector.erase(position)
```

- `position`: An iterator pointing to the element to be removed.

To eliminate a range of elements:

```
vector.erase(starting_position, ending_position)
```

- `starting_position`: An iterator indicating the beginning of the range to be removed.
- `ending_position`: An iterator indicating the end of the range to be removed.
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:27]: Modifications:
1. The above format maintains a structure for the reader.
2. Also, whenever we give a syntax, make sure to describe the parameters in short as above.

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+To use `.erase()`, the `vector` must be defined using `std::vector`.
+
+## Example 1
+
+In the example below *.erase()* removes a single element:
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:33]:
```suggestion
In the example below `.erase()` removes a single element:
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:33]: backticks to represent a term

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      std::cout << ' ' << room[i];
+    std::cout << '\n';
+
+     return 0;
+}
+```
+
+The output of the above code is :
+```shell
+Rooms available: 1 2 4 5 6 7 8 9 10
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:62]:
```suggestion
The output of the above code is :

```shell
Rooms available: 1 2 4 5 6 7 8 9 10
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:62]: one line space is needed

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+## Example 2
+
+In the example below *.erase()* removes a range of element:
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:67]:
```suggestion
In the example below `.erase()` removes a range of element:
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:67]: backticks to represent a term

File: content/cpp/concepts/vectors/terms/erase/erase.md (hunk 1 above)
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:94]:
```suggestion
The output of the above code is :

```shell
Rooms available: 1 2 4 5 6 7
```

Comment (mamtawardhani) [reply content/cpp/concepts/vectors/terms/erase/erase.md:94]: one line space is needed

File: content/cpp/concepts/vectors/terms/erase/erase.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    // Declare a vector
+    std::vector<int> room;      
+
+    // Add values (from 1 to 10)
+    for (int i=1; i<=10; i++) room.push_back(i); 
+
+    // Remove the 3rd element
+    room.erase (room.begin()+2);  
+    
+    // Print out element left in the vector
```
Comment (mamtawardhani) [inline content/cpp/concepts/vectors/terms/erase/erase.md:50]:
```suggestion
    // Print out elements left in the vector
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @Son0-lory, thank you for contributing to the Codecademy Docs, the entry is well written! 😃 

I've suggested a few changes, please review and modify them at your earliest convenience. Feel free to dicuss any point. Thank you! 😄

**Comment by Son0-lory** [comment]
Hi,
the code as been modified as suggested, thanks. 
Syntax paragrapher is now improved.

**Review by mamtawardhani — APPROVED** [review]
Thank you for making the required changes @Son0-lory 😃 

The entry looks good for a second round of review! 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @Son0-lory! 👍

**Pull request merged**